### ISSUE TITLE
Issue using multiple subdomains

### ISSUE THREAD

**Issue by githtz** [JrCs/docker-letsencrypt-nginx-proxy-companion#219]
I'm using this le-companion on a host (let's call it _example.com_) with multiple subdomains. I've been using the seperate container method to get my services online. 
My first service was nextcloud, with LETSENCRYPT_HOST and VIRTUAL_HOST set to  _nextcloud.example.com_. Everything works find here, certificates were created and are served correctly. 
Then I'd like to run a simple apache server as service. I created the service the same way as the nextcloud service, changed LETSENCRYPT_HOST and VIRTUAL_HOST to _web.example.com_.
When I try to access _web.example.com_ the certificates for _nextcloud.example.com_ are delivered. Did I mess something up?

I've set an A-Record for both _nextcloud.example.com_ and _web.example.com_ to the IP of my server.

Seems to be a problem with nginx, all subdomains are just some kind of alias to _nextcloud.example.com_. Could anyone point me into the right direction please? Thanks in advance!

**Comment by pidupuis** [comment]
Could you share us your docker-compose.yml file?

**Comment by githtz** [comment]
@pidupuis 
I haven't been using docker-compose, so here are the bash cmds I executed.
```
curl https://raw.githubusercontent.com/jwilder/nginx-proxy/master/nginx.tmpl > /var/docker-gen/templates/nginx.tmpl

docker run -d -p 80:80 -p 443:443 \
    --name nginx \
    --restart=always \
    -v /etc/nginx/conf.d  \
    -v /etc/nginx/vhost.d \
    -v /usr/share/nginx/html \
    -v /var/letsencrypt/certs:/etc/nginx/certs:ro \
    --label com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy=true \
    nginx

docker run -d \
    --name nginx-gen \
    --restart=always \
    --volumes-from nginx \
    -v /var/docker-gen/templates/nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl:ro \
    -v /var/run/docker.sock:/tmp/docker.sock:ro \
    -v /etc/nginx-gen/vhost.d:/etc/nginx/vhost.d:ro \
    jwilder/docker-gen \
    -notify-sighup nginx -watch -wait 5s:30s /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
    
docker run -d \
    --name nginx-letsencrypt \
    --restart=always \
    -e "NGINX_DOCKER_GEN_CONTAINER=nginx-gen" \
    --volumes-from nginx \
    -v /var/letsencrypt/certs:/etc/nginx/certs:rw \
    -v /var/run/docker.sock:/var/run/docker.sock:ro \
    jrcs/letsencrypt-nginx-proxy-companion


docker run -d --name=nextcloud \
    --restart=always \
    --link db:mysql \
    -v /var/nextcloud-data:/var/www/html \
    -e "VIRTUAL_HOST=nextcloud.example.com" \
    -e "LETSENCRYPT_HOST=nextcloud.example.com" \
    -e "LETSENCRYPT_EMAIL=me@mail.com" \
    nextcloud
    
docker run -d --name=db \
    --restart=always \
    -v /var/nextcloud-db:/var/lib/mysql \
    -e "MYSQL_ROOT_PASSWORD=secret_root_pass" \
    -e "MYSQL_PASSWORD=secret_user_pass" \
    -e "MYSQL_DATABASE=nextcloud" \
    -e "MYSQL_USER=nextcloud" \
    mariadb
```

Then I added the apache service via
```
docker run -d \
    --restart=always \
    --name=myapache \
    -v /var/www:/usr/local/apache2/htdocs/ \
    -e "VIRTUAL_HOST=web.example.com" \
    -e "LETSENCRYPT_HOST=web.example.com" \
    -e "LETSENCRYPT_EMAIL=me@mail.com" \
    httpd:2.4
```

**Comment by githtz** [comment]
Hey there,
I solved the problem. I found out that I am a complete idiot. 
By adding the line `-v /etc/nginx-gen/vhost.d:/etc/nginx/vhost.d:ro` when calling _jwilder/docker-gen_ I prevented the le-companion from enabling public access to my subdomains .well-known folders. After I removed the parameter everything worked fine.

**Comment by epos** [comment]
For further reference: I had the same problem when hosting multiple subdomains in the same container. This was solved using one container for each domain.


What doesn't work (for me):

    "LETSENCRYPT_HOST=subdomain1.example.com,subdomain2.example.com"

**Comment by trainingcity** [comment]
I'm having a problem with a CNAME DNS entry for www. The DNS redirects to the @ IP address but the Nginx container responds with a 503.  I don't want to run a separate container for @ and www.

**Comment by buchdag** [comment]
Have you configured your proxyed container with `VIRTUAL_HOST=yourdomain.com,www.yourdomain.com` ?

**Comment by trainingcity** [comment]
Thanks @buchdag. I edited a  .env file referenced in a docker-compose.yml file to include the www.magurocafe.com where it had only included magurocafe.com:

VIRTUAL_HOST=magurocafe.com,www.magurocafe.com
LETSENCRYPT_HOST=magurocafe.com,www.magurocafe.com

I tried restarting all the containers (nginx-proxy, nginx-proxy-gen, nginx-proxy-le, and the containers running wordpress/mysql.

It still gave the same 503 error so I also tried running the following docker command:

docker exec nginx-proxy-le /app/force_renew

As everything is using production lets encrypt, I don't want to run that command too many times.

Any ideas welcome!

**Comment by buchdag** [comment]
Could you dump the content of `/etc/nginx/conf.d/default.conf` and the result of `docker inspect` on your proxied wordpress container (please remove or anonymise any contact email present in the `inspect` result) ?

**Comment by trainingcity** [comment]
The wp container is using apache. The apache2.conf and other conf files seem to be in default.  I wonder if I should map a virtual host config to the container?

Just an FYI, the service works great for https://magurocafe.com and for several other containers running behind the nginx-proxy setup. None of the other containers have a CNAME pointing to their associated @.

I noticed that the force renew command returned magurocafe.com but not www.magurocafe.com 

Here's the output from docker inspect of the container. There is no reference to www.magurocafe.com:

[
    {
        "Id": "7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb",
        "Created": "2019-06-07T15:23:29.08862267Z",
        "Path": "docker-entrypoint.sh",
        "Args": [
            "apache2-foreground"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 9398,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-07-11T14:16:31.516883562Z",
            "FinishedAt": "2019-07-11T14:16:11.134542671Z"
        },
        "Image": "sha256:35371c52b7d448444004a351ae4e0a46981a65ca6656c1dd5948b0728c16cbfc",
        "ResolvConfPath": "/var/lib/docker/containers/7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb/hostname",
        "HostsPath": "/var/lib/docker/containers/7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb/hosts",
        "LogPath": "/var/lib/docker/containers/7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb/7d168bb983869ecf07c9882a0ed0b8607d06bd3b66ca3e35dd9d896695f15dfb-json.log",
        "Name": "/maguro-wp",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": [
            "6d59ce5ecfa3c3a83311bd4a2f1b7cfec89f8f911e89b35b9c5f04847fa30b64",
            "2bb63d5c97c4f31967209be479a39d538db8dc7774ec09790ac674e86eddb66c"
        ],
        "HostConfig": {
            "Binds": [
                "#########/themes:/var/www/html/wp-content/themes:rw",
                "#########/magurocafe/custom/uploads.ini:/usr/local/etc/php/conf.d/uploads.ini:rw",
                "#########/magurocafe/plugins:/var/www/html/wp-content/plugins:rw",
                "#########/magurocafe/var/www/html:/var/www/html:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "nginx-proxy",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": [],
            "CapAdd": null,
            "CapDrop": null,
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/61585b69cd3f81250a20bb63c7bd7aae03c59e6a326598b3b3471e49c29d3dea-init/diff:/var/lib/docker/overlay2/9a77342cef27d4c18f25cd2a7961090907698ec28e265474419eb356f46ee8ef/diff:/var/lib/docker/overlay2/6c31ac838afe1412742edd439c6324c4c9257bf9b09f6299aa8ee99c7f5f8746/diff:/var/lib/docker/overlay2/d9241503370b2cf0dda5d836e0da17d6de8f5ddde17d18d4272b319d57303efa/diff:/var/lib/docker/overlay2/0d7a7bd4c9d91ae34f5ce65798d658f9d067c1cd91f65ee55faefb979e20ab23/diff:/var/lib/docker/overlay2/a6f8aae7577d684a948958b994984fb0a039070cc90192416674cde5b8ca2f3f/diff:/var/lib/docker/overlay2/e770c8a8d22e00855549e5724006881bde8097b6fc2fc84f1885cfd4beae2eb6/diff:/var/lib/docker/overlay2/6877b725029670534d5be817b7826b3ae97084fe2d98cbaab4dd1b01c1c7bdc7/diff:/var/lib/docker/overlay2/38b40e594ac2c45358329ced923dc7866567075e385d8cf397cc5b9f4526aaa0/diff:/var/lib/docker/overlay2/01def2a21cc3062f2f3ef1b29cb35ebf37c72960827cfde00c372db688fa7f14/diff:/var/lib/docker/overlay2/78ab9233a522f28c712622625826391cdccb29de3945645dbd1303aa2d853862/diff:/var/lib/docker/overlay2/f37434925392234f55adfd63c7673cf077c411ba3764f9223faef6f5f5b2c7f9/diff:/var/lib/docker/overlay2/79c6ac9035de576861da402b0847c4b731525eb5c7e790e4b2104f880c1bd58a/diff:/var/lib/docker/overlay2/beea844c483e231eb82f2dafd3070e25196fff6b9958bdb1811facdb3f0e4984/diff:/var/lib/docker/overlay2/e394c9cc7b3f582f1c17c5d1d74fcb5e6e7d342c9bc1b4ce97fb71cd4c8ee08b/diff:/var/lib/docker/overlay2/03550864a685525292b9cc5d9e7a6f8c8944ae0e9dd65253d44bec638fd2db0b/diff:/var/lib/docker/overlay2/812e08fd1465391ea5e229208096d56ccab867be8a6431f4e3aace883668806b/diff:/var/lib/docker/overlay2/9602b4911f3d1000f1fd6a67e3d386e830c355bef162467eba6172c12e1e92ca/diff:/var/lib/docker/overlay2/4ee45c5ac05d530ff1ab7361e33e3e4748e2a3f508502451ea7b2cdf39447378/diff:/var/lib/docker/overlay2/c6d1db22712e8a0520183ffc076e9e1fc1651c4815788fa8a117983c056e7cbd/diff:/var/lib/docker/overlay2/4212cce28cd231e0784e88c4b47c3e818902a3d83bf334ccfeba3f44183e0dc5/diff",
                "MergedDir": "/var/lib/docker/overlay2/61585b69cd3f81250a20bb63c7bd7aae03c59e6a326598b3b3471e49c29d3dea/merged",
                "UpperDir": "/var/lib/docker/overlay2/61585b69cd3f81250a20bb63c7bd7aae03c59e6a326598b3b3471e49c29d3dea/diff",
                "WorkDir": "/var/lib/docker/overlay2/61585b69cd3f81250a20bb63c7bd7aae03c59e6a326598b3b3471e49c29d3dea/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "#########/magurocafe/var/www/html",
                "Destination": "/var/www/html",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/#########/magurocafe/themes",
                "Destination": "/var/www/html/wp-content/themes",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "#########/magurocafe/custom/uploads.ini",
                "Destination": "/usr/local/etc/php/conf.d/uploads.ini",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/#########/magurocafe/plugins",
                "Destination": "/var/www/html/wp-content/plugins",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "7d168bb98386",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "MYSQL_ROOT_PASSWORD=########",
                "MYSQL_DATABASE=########",
                "MYSQL_USER=#######",
                "MYSQL_PASSWORD=##########",
                "WORDPRESS_DB_HOST=maguro-mysql:3306",
                "WORDPRESS_DB_USER=######",
                "WORDPRESS_DB_PASSWORD=#######",
                "WORDPRESS_DB_NAME=#######",
                "VIRTUAL_HOST=magurocafe.com",
                "LETSENCRYPT_HOST=magurocafe.com",
                "LETSENCRYPT_EMAIL=#########",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkg-config \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "APACHE_CONFDIR=/etc/apache2",
                "APACHE_ENVVARS=/etc/apache2/envvars",
                "PHP_EXTRA_BUILD_DEPS=apache2-dev",
                "PHP_EXTRA_CONFIGURE_ARGS=--with-apxs2 --disable-cgi",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2",
                "PHP_LDFLAGS=-Wl,-O1 -Wl,--hash-style=both -pie",
                "GPG_KEYS=1729F83938DA44E27BA0F4D3DBDB397470D12172 B1B44D8F021E4E2D6021E995DC9FF8D3EE5AF27F",
                "PHP_VERSION=7.2.13",
                "PHP_URL=https://secure.php.net/get/php-7.2.13.tar.xz/from/this/mirror",
                "PHP_ASC_URL=https://secure.php.net/get/php-7.2.13.tar.xz.asc/from/this/mirror",
                "PHP_SHA256=14b0429abdb46b65c843e5882c9a8c46b31dfbf279c747293b8ab950c2644a4b",
                "PHP_MD5=",
                "WORDPRESS_VERSION=5.0.2",
                "WORDPRESS_SHA1=4a6971d35eb92e2fc30034141b1c865e8c156add"
            ],
            "Cmd": [
                "apache2-foreground"
            ],
            "ArgsEscaped": true,
            "Image": "magurocafe_maguro-wp",
            "Volumes": {
                "/usr/local/etc/php/conf.d/uploads.ini": {},
                "/var/www/html": {},
                "/var/www/html/wp-content/plugins": {},
                "/var/www/html/wp-content/themes": {}
            },
            "WorkingDir": "/var/www/html",
            "Entrypoint": [
                "docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "#########",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "magurocafe",
                "com.docker.compose.service": "maguro-wp",
                "com.docker.compose.slug": "#########9",
                "com.docker.compose.version": "1.23.1"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "#########",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "80/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/0a173eaecb81",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "nginx-proxy": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "7d168bb98386",
                        "maguro-wp"
                    ],
                    "NetworkID": "#########",
                    "EndpointID": "1#########0",
                    "Gateway": "172.25.0.1",
                    "IPAddress": "172.25.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:19:00:04",
                    "DriverOpts": null
                }
            }
        }
    }
}

**Comment by buchdag** [comment]
> Here's the output from docker inspect of the container. There is no reference to www.magurocafe.com:

Your config is incorrect and your container does not inherit the environment variable it should.

**Comment by trainingcity** [comment]
Thanks @buchdag. Can you see what is wrong with the config?  btw, here is the nginx/conf.d/default.conf file minus most of the non related containers except one for reference:


root@12a2efa1ebce:/etc/nginx/conf.d# cat default.conf
# If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
# scheme used to connect to this server
map $http_x_forwarded_proto $proxy_x_forwarded_proto {
  default $http_x_forwarded_proto;
  ''      $scheme;
}
# If we receive X-Forwarded-Port, pass it through; otherwise, pass along the
# server port the client connected to
map $http_x_forwarded_port $proxy_x_forwarded_port {
  default $http_x_forwarded_port;
  ''      $server_port;
}
# If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
# Connection header that may have been passed to this server
map $http_upgrade $proxy_connection {
  default upgrade;
  '' close;
}
# Apply fix for very long server names
server_names_hash_bucket_size 128;
# Default dhparam
# Set appropriate X-Forwarded-Ssl header
map $scheme $proxy_x_forwarded_ssl {
  default off;
  https on;
}
gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
log_format vhost '$host $remote_addr - $remote_user [$time_local] '
                 '"$request" $status $body_bytes_sent '
                 '"$http_referer" "$http_user_agent"';
access_log off;
# HTTP 1.1 support
proxy_http_version 1.1;
proxy_buffering off;
proxy_set_header Host $http_host;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $proxy_connection;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
proxy_set_header X-Forwarded-Ssl $proxy_x_forwarded_ssl;
proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
# Mitigate httpoxy attack (see README for details)
proxy_set_header Proxy "";
server {
        server_name _; # This is just an invalid value which will never trigger on a real hostname.
        listen 80;
        access_log /var/log/nginx/access.log vhost;
        return 503;
}
server {
        server_name _; # This is just an invalid value which will never trigger on a real hostname.
        listen 443 ssl http2;
        access_log /var/log/nginx/access.log vhost;
        return 503;
        ssl_session_tickets off;
        ssl_certificate /etc/nginx/certs/default.crt;
        ssl_certificate_key /etc/nginx/certs/default.key;
}
# magurocafe.com
upstream magurocafe.com {
                                ## Can be connected with "nginx-proxy" network
                        # maguro-wp
                        server 172.25.0.4:80;
                                ## Can be connected with "nginx-proxy" network
                # maguro-mysql
                        server 172.25.0.2 down;
}
server {
        server_name magurocafe.com;
        listen 80 ;
        access_log /var/log/nginx/access.log vhost;
        return 301 https://$host$request_uri;
}
server {
        server_name magurocafe.com;
        listen 443 ssl http2 ;
        access_log /var/log/nginx/access.log vhost;
        ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
        ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:!DSS';
        ssl_prefer_server_ciphers on;
        ssl_session_timeout 5m;
        ssl_session_cache shared:SSL:50m;
        ssl_session_tickets off;
        ssl_certificate /etc/nginx/certs/magurocafe.com.crt;
        ssl_certificate_key /etc/nginx/certs/magurocafe.com.key;
        ssl_dhparam /etc/nginx/certs/magurocafe.com.dhparam.pem;
        ssl_stapling on;
        ssl_stapling_verify on;
        ssl_trusted_certificate /etc/nginx/certs/magurocafe.com.chain.pem;
        add_header Strict-Transport-Security "max-age=31536000" always;
        include /etc/nginx/vhost.d/default;
        location / {
                proxy_pass http://magurocafe.com;
        }
}
# pma.phone.yoga
upstream pma.phone.yoga {
                                ## Can be connected with "nginx-proxy" network
                        # phpmyadmin
                        server 172.25.0.9:80;
}
server {
        server_name pma.phone.yoga;
        listen 80 ;
        access_log /var/log/nginx/access.log vhost;
        return 301 https://$host$request_uri;
}
server {
        server_name pma.phone.yoga;
        listen 443 ssl http2 ;
        access_log /var/log/nginx/access.log vhost;
        ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
        ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:!DSS';
        ssl_prefer_server_ciphers on;
        ssl_session_timeout 5m;
        ssl_session_cache shared:SSL:50m;
        ssl_session_tickets off;
        ssl_certificate /etc/nginx/certs/pma.phone.yoga.crt;
        ssl_certificate_key /etc/nginx/certs/pma.phone.yoga.key;
        ssl_dhparam /etc/nginx/certs/pma.phone.yoga.dhparam.pem;
        ssl_stapling on;
        ssl_stapling_verify on;
        ssl_trusted_certificate /etc/nginx/certs/pma.phone.yoga.chain.pem;
        add_header Strict-Transport-Security "max-age=31536000" always;
        include /etc/nginx/vhost.d/default;
        location / {
                proxy_pass http://pma.phone.yoga;
        }
}
..../

**Comment by buchdag** [comment]
> Can you see what is wrong with the config?

Nope, this depends on your compose and .env files, and I won't be able to help you with this as it's not directly related to `letsencrypt-nginx-proxy-companion`.

Please check the Docker Compose documentation.

**Comment by trainingcity** [comment]
Got it, thanks. I'll review the wp container config and try to see how it gets the virtual host values.

**Comment by trainingcity** [comment]
Ok, for anyone who reads this threat. 

I ended up just creating a second identical wordpress container with a new name in the same docker-compose.yml file. The second wordpress container refers to the same persistent volumes and the same mysql db container. The only difference is the environmental values. I pointed it to a different env_file that has two different entries for virtual host and lets encrypt host. Instead of magurocafe.com, they are set to www.magurocafe.com.
